examples/sql_bank/main.go

```diff
@@ -53,17 +53,10 @@ func moveMoney(db *sql.DB) {
 
 		switch *transferStyle {
 		case "single-stmt":
-			// TODO(pmattis): Need to add a sub-SELECT here so that we don't create
-			// negative balances:
-			//
-			//   WHERE id IN ($1, $2) AND (SELECT balance >= $3 FROM bank.accounts WHERE id = $1)
-			//
-			// This doesn't currently work as we don't expand placeholders in
-			// sub-SELECTs correctly.
 			update := `
 UPDATE bank.accounts
   SET balance = CASE id WHEN $1 THEN balance-$3 WHEN $2 THEN balance+$3 END
-  WHERE id IN ($1, $2)
+  WHERE id IN ($1, $2) AND true IN (SELECT balance >= $3 FROM bank.accounts WHERE id = $1)
```

Comment:
I think this approach is incorrect for the purpose of this example, because it assumes that a query isn't necessary to find two valid account numbers. We should first query to find two suitable accounts to avoid this assumption.